services/src/main/java/org/exoplatform/officeonline/webui/OfficeOnlineContext.java

```diff
@@ -0,0 +1,228 @@
+   */
+  private OfficeOnlineContext(WebuiRequestContext requestContext) throws OfficeOnlineException {
+    JavascriptManager js = requestContext.getJavascriptManager();
+    this.require = js.require("SHARED/officeonline", "officeonline");
+
+    // Basic JS module initialization
+    String messagesJson;
+    try {
+      ResourceBundleService i18nService = requestContext.getApplication()
+                                                        .getApplicationServiceContainer()
+                                                        .getComponentInstanceOfType(ResourceBundleService.class);
+      ResourceBundle res = i18nService.getResourceBundle("locale.officeonline.OfficeOnlineClient", requestContext.getLocale());
+      Map<String, String> resMap = new HashMap<String, String>();
+      for (Enumeration<String> keys = res.getKeys(); keys.hasMoreElements();) {
+        String key = keys.nextElement();
+        String bundleKey;
+        if (key.startsWith(CLIENT_RESOURCE_PREFIX)) {
+          bundleKey = key.substring(CLIENT_RESOURCE_PREFIX.length());
+        } else {
+          bundleKey = key;
+        }
+        resMap.put(bundleKey, res.getString(key));
+      }
+      messagesJson = new JsonGeneratorImpl().createJsonObjectFromMap(resMap).toString();
+    } catch (JsonException e) {
+      LOG.warn("Cannot serialize messages bundle JSON", e);
+      messagesJson = "{}";
+    } catch (Exception e) {
+      LOG.warn("Cannot build messages bundle", e);
+      messagesJson = "{}";
+    }
+
+    ConversationState convo = ConversationState.getCurrent();
+    if (convo != null && convo.getIdentity() != null) {
+      ExoContainer container = requestContext.getApplication().getApplicationServiceContainer();
+      CometdOfficeOnlineService cometdService = container.getComponentInstanceOfType(CometdOfficeOnlineService.class);
+
+      String userId = convo.getIdentity().getUserId();
+
+      CometdConfig cometdConf = new CometdConfig(cometdService.getCometdServerPath(),
+                                                 cometdService.getUserToken(userId),
+                                                 PortalContainer.getCurrentPortalContainerName());
+      try {
+        callOnModule("init('" + userId + "', " + cometdConf.toJSON() + ", " + messagesJson + ");");
+      } catch (JsonException e) {
+        LOG.warn("Cannot create JSON from cometd config: {}", e.getMessage());
+        throw new OfficeOnlineException("Failed to convert cometd config to JSON", e);
+      }
+    } else {
+      throw new OfficeOnlineException("Authenticated user required");
+    }
+  }
+
+  /**
+   * App require JS.
+   *
+   * @return the require JS
+   */
+  private RequireJS appRequireJS() {
+    return require;
+  }
+
+  /**
+   * Call on module.
+   *
+   * @param code the code
+   */
+  private void callOnModule(String code) {
+    require.addScripts(new StringBuilder("officeonline.").append(code).append("\n").toString());
```

Comment:
Doing this way we "a bit hide" actual logic of what is happening. This looks smart and code reuse here in the code, but readability degrades in general. And if for some reason we'll need invoke something on another AMD or just a script this helper function will not be useful but used together with direct call to `require` will confuse like it's different things. I propose don't do this and let the code be well readable what is good in long term maintenance.

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/22/commits/a9c4c730e930d0319b18dac2033420e418dd6290